submissions/assmass13/port-sniffer/sniffer.js

```diff
@@ -0,0 +1,127 @@
+const dns = require('dns');
+const { Socket } = require('net');
+
+const defaultPortLimits = {
+  startPort: 0,
+  endPort: 65535
+};
+const timeout = 300;
+
+const helpMessage = `Sniffer params usage:
+--host - required argument which can be either domain name, like google.com or IP address like 172.217.3.110
+--ports - limits range of ports to scan, should be provided in format (start_port)-(end_port), for instance: 3-600
+--help - flag to see hint about how to use TCP sniffer\n`;
+const hostErrorMessage = `--host parameter with its value should be passed\n`;
+const portsErrorMessage = `--ports value should be passed as a port range with limits(0-65535), like 15-333\n`;
+function lookupErrorMessage(error) {
+  return `Error ${error.code} occurred during ${error.syscall} call for '${error.hostname}' host\n`;
+}
+
+function stdoutWrite(str) {
+  process.stdout.write(str);
+}
+
+function exit(code) {
+  process.exit(code);
+}
+
+function parseArgs(args) {
+  if (args.includes('--help')) {
+    stdoutWrite(helpMessage);
+    exit(0);
+  } else if (
+    args.indexOf('--host') === -1 ||
+    !args[args.indexOf('--host') + 1]
+  ) {
+    stdoutWrite(hostErrorMessage);
+    exit(1);
+  }
+  return args.reduce((result, item, index, arr) => {
+    const newResult = { ...result };
```

Comment:
Why do you create new object on every iteration?

Reply:
https://eslint.org/docs/2.0.0/rules/no-param-reassign

Reply:
Oh, I see. Here is how I would tackle this:
```js
  return args.reduce((result, item, index, arr) => {
    if (item.startsWith('--') && arr[index + 1]) {
      return { ...result, [item]: arr[index + 1] };
    }
    return result;
  }, {});
```
Just a suggestion.

Reply:
fixed